hdf5-sys/build.rs

```diff
@@ -111,17 +116,20 @@ fn validate_runtime_version(config: &Config) {
     let libfiles = &["libhdf5.dylib", "libhdf5.so", "hdf5.dll"];
     let mut link_paths = config.link_paths.clone();
     if cfg!(all(unix, not(target_os = "macos"))) {
-        if let Some(ldv) = run_command("ld", &["--verbose"]) {
-            let re = Regex::new(r#"SEARCH_DIR\("=?(?P<path>[^"]+)"\)"#).unwrap();
-            println!("Adding extra link paths (ld)...");
-            for caps in re.captures_iter(&ldv) {
-                let path = &caps["path"];
-                println!("    {}", path);
-                link_paths.push(path.into());
-            }
-        } else {
-            println!("Unable to add extra link paths (ld).");
-        }
+        run_command("ld", &["--verbose"]).map_or_else(
```

Comment:
Tbh, these `.map_or_else()` actually make it much less readable :( Would clippy be against a simple match as well? Maybe we can disable this particular lint?